src/main/java/com/amazon/ion/impl/BufferedOutputStream.kt

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl
+
+import com.amazon.ion.impl.bin.*
+import java.io.OutputStream
+
+/**
+ * TODO:
+ *   - Add proper tests
+ */
+internal class BufferedOutputStream(
```

Comment:
🗺️ This allows the text writer to be buffered, like the binary writer, so that we can make sure that we buffer enough before writing out the encoding directives. We still need to add some way of specifying parts of the encoding directives ahead of time so that we don't have to do any buffering. If we can guarantee that there will be no unexpected encoding directives, then we don't need to buffer the text writer.